typing/ctype.ml

```diff
@@ -567,39 +576,62 @@ let really_closed = ref None
    [free_variables] drops the type/row information
    and only returns a [variable list].
  *)
-let rec free_vars_rec real ty =
+let rec free_vars_rec id_pairs real ty =
   let ty = repr ty in
   if try_mark_node ty then
     match ty.desc, !really_closed with
       Tvar _, _ ->
         free_variables := (ty, real) :: !free_variables
     | Tconstr (path, tl, _), Some env ->
         begin try
-          let (_, body, _) = Env.find_type_expansion path env in
+          let (_, body, _) =
+            Env.find_type_expansion (Path.subst id_pairs path) env
```

Comment:
`id_pairs` will only contain ids coming from the `Tfunctor` node that `free_vars_rec` will have crossed, nothing from "the outside" is reaching this point.
So I think you could just add the original (unscoped) path to the environment. Or do you think that'd cause some issues later on?

Reply:
I would prefer if 'no unscoped identifiers in the environment' was an invariant that we never break. I think that it's also possible that this could cause a `Ident.No_scope` exception (while attempting to set up a rescope substitution in the environment) in the current code.